### PR TITLE
Lint on toggle

### PR DESCRIPTION
**Pull request by shawninder**
solves this issue: https://github.com/atom-community/linter/issues/894#issuecomment-141783902

### CODE REVIEW

**Comment by steelbrain** [comment]
A changelog entry would be appreciated along with a spec.

**Comment by shawninder** [comment]
I don't really know how to test this... I don't even know how to run the tests. There's no `npm test` script...

**Comment by steelbrain** [comment]
@shawninder `apm test` from the command line, or open this project in atom and run `Run Package Specs` from command palette

**Comment by shawninder** [comment]
@steelbrain  Is `Sinon` available in the spec files? I'd like to create a spy on the `lint` function...

**Comment by steelbrain** [comment]
@shawninder I don't think it is, we have jasmine 1.X. Here's what you can do to test it, register a provider, make it's lint function set a variable to true, toggle linters, toggle again, validate that the variable is set to true

**Comment by shawninder** [comment]
Hmm, jasmine does have spies, so the following should be possible.

```it 'lints on toggle', ->
      spyOn(linter, 'lint')
      linter.commands.togglePanel()
      linter.commands.togglePanel()
      expect(linter.lint).toHaveBeenCalled()```

But I get `lint() method does not exist`

**Comment by shawninder** [comment]
Ah, it's `linter.commands.lint`, sorry

**Comment by shawninder** [comment]
But the test fails (yet manual testing shows that it works...)

```
it 'lints on toggle', ->
      spyOn(linter.commands, 'lint')
      linter.commands.togglePanel()
      linter.commands.togglePanel()
      expect(linter.commands.lint).toHaveBeenCalled()
```

**Comment by steelbrain** [comment]
I'll try it locally and post my findings

**Comment by shawninder** [comment]
I suspect it might be a timing issue, but async testing doesn't seem to be working as the following test passes (notice the `expect(true).toBy(false)`):

```
it 'lints on toggle', (done) ->
      spyOn(linter.commands, 'lint')
      linter.commands.togglePanel()
      linter.commands.togglePanel()
      setTimeout(->
        expect(linter.commands.lint).toHaveBeenCalled()
        expect(true).toBe(false)
        done()
      , 100)
```

**Comment by shawninder** [comment]
ok, I was using new jasmine syntax. Very old jasmine (1.x) looks more like this:

```
it 'lints on toggle', ->
      flag = false
      runs(->
        spyOn(linter.commands, 'lint')
        linter.commands.togglePanel()
        linter.commands.togglePanel()
        setTimeout(->
          flag = true
          expect(linter.commands.lint).toHaveBeenCalled()
        , 500)
      )
      waitsFor(->
        return flag
      , '`linter.commands.lint` should have been called', 500)
```
But this test times out

**Comment by steelbrain** [comment]
Well, how about this?
```diff
diff --git a/spec/commands-spec.coffee b/spec/commands-spec.coffee
index 152c4ff..d594c98 100644
--- a/spec/commands-spec.coffee
+++ b/spec/commands-spec.coffee
@@ -13,3 +13,11 @@ describe 'Commands', ->
       expect(linter.views.panel.getVisibility()).toBe(!visibility)
       linter.commands.togglePanel()
       expect(linter.views.panel.getVisibility()).toBe(visibility)
+  describe 'linter:toggle', ->
+    it 'relint when enabled', ->
+      waitsForPromise ->
+        atom.workspace.open(__dirname + '/fixtures/file.txt').then ->
+          spyOn(linter.commands, 'lint')
+          linter.commands.toggleLinter()
+          linter.commands.toggleLinter()
+          expect(linter.commands.lint).toHaveBeenCalled()
```

**Comment by shawninder** [comment]
yeah that works, are you pushing this or shall I do it?

**Comment by steelbrain** [comment]
It would be better if you do

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 * Fixed messages so that they line up and don't wrap in weird ways (Fixes #859)
 
+## 1.5.3
```
Comment (steelbrain) [inline CHANGELOG.md]: You should put this entry on the upcoming section

File: package.json [hunk 2]
```diff
@@ -2,7 +2,7 @@
   "name": "linter",
   "main": "./lib/main",
   "author": "atom-community",
-  "version": "1.5.2",
+  "version": "1.5.3",
```
Comment (steelbrain) [inline package.json]: We'll bump this version automatically, don't bother :)

**Comment by shawninder** [comment]
done

**Comment by shawninder** [comment]
ok, never used changelogs before. Done now

**Comment by steelbrain** [comment]
Travis is stuck for some reasons :sweat_smile: 
Just waiting on CI before merging

**Comment by shawninder** [comment]
yeah Amazon's servers are down so travis is going to take a while. Thanks for helping me through this @steelbrain

**Comment by steelbrain** [comment]
Look good to me. Thanks for your awesome contribution :clap:

**Comment by shawninder** [comment]
Awesome may be pushing it a little ;) Any ETA on when a new version will be pushed?

**Comment by steelbrain** [comment]
@shawninder just a few more features or a few day.

**Comment by shawninder** [comment]
Beautiful

**Pull request merged**